### ISSUE TITLE
kube-inject failed for helm install

### ISSUE THREAD

**Issue by linsun** [istio/istio#3182]
when auto sidecar injector is disabled:

```
$ kubectl create -f <(istioctl kube-inject -f  ../../samples/bookinfo/kube/bookinfo.yaml)
Error: could not read valid configmap "istio" from namespace  "istio-system": configmaps "istio" not found - Use --meshConfigFile or re-run kube-inject with `-i <istioSystemNamespace> and ensure valid MeshConfig exists
error: no objects passed to create
```

Discussed with @ayj -- we should be installing the istio config map regardless if the side car injector is enabled or not.  

cc @luisyonaldo @sdake

**Comment by linsun** [comment]
@luisyonaldo I think we can create a istio-config under helm/istio/charts dir and move the configmap from sidecar-injector/templates to there.  WDYT?  curious if you have any ideas?

**Comment by ayj** [comment]
There are two configmaps: `istio` and `istio-inject`. `istio` is required by other components and should always be created, e.g. Pilot (see https://github.com/istio/istio/blob/31c96273820648af23478f10b123bf4d1a759ba3/install/kubernetes/helm/istio/charts/pilot/templates/deployment.yaml#L82). 

The `istio-inject` is required for automatic sidecar injection and also useful for manual injection if uses want to use the `--meshConfigMapName` option.

**Comment by luisyonaldo** [comment]
they can both be created inside the same configmap since both use different filenames, `mesh` and `config`, but not sure what would be the benefits? 

Right now, helm re creates deployments if configmap changes, if both use same configmap, a change for istio inject would also trigger a recreation of the other components that use same configmap.

**Comment by linsun** [comment]
ok looks like I did have the istio config map installed but it is not called "istio".  Unfortunately it needs to be called istio because istioctl will look up the istio config map when no mesh config param is supplied during kibe-inject.  

```
$ kubectl get configmaps --namespace=istio-system
NAME                                    DATA      AGE
istio-ingress-controller-leader-istio   0         1h
istio-istio-mesh-config                 1         1h
istio-mixer                             1         1h
istio-mixer-custom-resources            1         1h
```

**Comment by luisyonaldo** [comment]
is it possible to pass this a parameter? right now all names in helm charts include `release` (so is dynamic) so you can have multiple version of the same chart without any issue.

**Comment by linsun** [comment]
yes, you can supply the --meshConfigFile to use your own mesh config if you choose.  however, the default one it reads is istio at the moment.  cc @ayj

**Comment by luisyonaldo** [comment]
can I add it somewhere [here](https://github.com/istio/istio/blob/master/install/kubernetes/helm/istio/charts/pilot/templates/deployment.yaml#L58-L74)?

**Comment by linsun** [comment]
as the args to the proxy image?  I don't think that would work....  it is an part of `istioctl kube-inject` right now:

**Comment by luisyonaldo** [comment]
well hardcoding the name will not be really bad since it only mean you can't have multiple istio helm versions running in the same namespace, since all will try to create same configmap in the same namespace. But I don't think there is an scenario where you may want to have multiple istio installations on the same namespace

**Comment by linsun** [comment]
agreed, we definitely don't want people to have multiple istio in the same namespace.  In fact, multiple istio installs where each in its own namespaces are not recommended either.

**Comment by luisyonaldo** [comment]
Ok then all dynamic naming is not needed

**Comment by linsun** [comment]
fixed in #3187
